offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/projects/ProjectDbConnector.groovy

```diff
@@ -0,0 +1,190 @@
+
+package life.qbic.portal.offermanager.dataresources.projects
+
+import groovy.util.logging.Log4j2
+import life.qbic.datamodel.dtos.business.Customer
+import life.qbic.datamodel.dtos.business.ProjectManager
+import life.qbic.datamodel.dtos.general.Person
+import life.qbic.portal.offermanager.dataresources.persons.PersonDbConnector
+
+import life.qbic.datamodel.dtos.projectmanagement.*
+
+import life.qbic.business.exceptions.DatabaseQueryException
+import life.qbic.portal.offermanager.dataresources.database.ConnectionProvider
+
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.sql.Statement
+
+/**
+ * Provides operations on QBiC project data
+ *
+ * This class is responsible for transferring project data to the customer db
+ *
+ * @since: 1.0.0
+ *
+ */
+@Log4j2
+class ProjectDbConnector {
+
+  /**
+   * A connection to the customer database used to create queries.
+   */
+  private final ConnectionProvider connectionProvider
+  private final PersonDbConnector personDbConnector
+
+  /**
+   * Constructor for a ProjectDbConnector
+   * @param connectionProvider a connection provider to the customer db
+   * @param personDbConnector db connector used to connect projects to customer and manager
+   * 
+   */
+  ProjectDbConnector(ConnectionProvider connectionProvider, PersonDbConnector personDbConnector) {
+    this.connectionProvider = connectionProvider
+    this.personDbConnector = personDbConnector
+  }
+  
+  /**
+   * parses existing projects from user database, might be needed later if more complex information is to be listed
+   */
+  public List<ProjectIdentifier> fetchProjects() {
+    List<ProjectIdentifier> projects = []
+    String query = "SELECT openbis_project_identifier from projects"
+    Connection connection = connectionProvider.connect()
+    connection.withCloseable {
+      def preparedStatement = it.prepareStatement(query)
+      ResultSet resultSet = preparedStatement.executeQuery()
+      while(resultSet.next()) {
+        try {
+          
+        }
+        String[] tokens = resultSet.getString('openbis_project_identifier').split("/")
+        ProjectSpace space = tokens[1]
+        ProjectCode project = tokens[2]
+        catch (Exception e) {
+          e.printStackTrace()
+          throw new DatabaseQueryException("Could not parse existing projects from database.")
+        }
+        projects.add(new ProjectIdentifier(space, project))
+      }
+    }
+    return projects
+  }
+
+  public Project addProjectAndConnectPersonsInUserDB(projectIdentifier, projectApplication) {
```

Comment:
Still JavaDocs missing

Reply:
I screwed up the base branch, I'll fix it with a merge